core/src/main/java/de/danoeh/antennapod/core/feed/FeedPreferences.java

```diff
@@ -58,11 +68,13 @@ public static FeedPreferences fromCursor(Cursor cursor) {
         boolean autoRefresh = cursor.getInt(indexAutoRefresh) > 0;
         int autoDeleteActionIndex = cursor.getInt(indexAutoDeleteAction);
         AutoDeleteAction autoDeleteAction = AutoDeleteAction.values()[autoDeleteActionIndex];
+        int volumeReductionIndex = cursor.getInt(indexVolumeReduction);
+        VolumeReductionSetting volumeReductionSetting = VolumeReductionSetting.values()[volumeReductionIndex];
```

Comment:
Please save a string (eg name) instead. It is too easy to unknowingly change the order of the enum items.

Reply:
You're right, I'll do that

Reply:
On second thought: I already did change the naming (e.g. strong/high -> heavy), and there might still be better terms. The order, however, seems natural. Volume reduction is either off, or set to an increasingly high level. To account for that, I might save an identifier like `level0`, `level1`, and so on, and then map them on the Enum by its ordinal - which might fall victim to uninformed refactoring, true.

It's just a feeling, I'm alright with both ways. I'll leave the decision up to you, @ByteHamster, and try to implement it shortly

Reply:
Maybe you could add a method to the enum that allows to map the number to the enum. That way, when refactoring, it is closer and therefore will less likely be overlooked. Maybe you can even add a parameter to the enum, so that the number is explicitly stated.


Something like this (question is kind of unrelated, just look at the sample code):
https://stackoverflow.com/a/36010298/4193263